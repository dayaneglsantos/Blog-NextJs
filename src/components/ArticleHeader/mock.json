{
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "title": "TÃ­tulo da postagem",
  "excerpt": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero ratione quisquam alias doloribus veniam dolore. Ea temporibus illum numquam ab quo.",
  "image": "/image.com",
  "author": {
    "slug": "dayane-santos",
    "displayName": "Dayane Santos"
  },
  "categories": [
    {
      "slug": "react",
      "displayName": "React"
    },
    {
      "slug": "js",
      "displayName": "Javascript"
    }
  ]
}
