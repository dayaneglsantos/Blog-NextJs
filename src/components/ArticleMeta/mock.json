{
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "darkTheme": false,
  "author": {
    "slug": "dayane-santos",
    "displayName": "Dayane Santos"
  },
  "categories": [
    {
      "slug": "react",
      "displayName": "React"
    },
    {
      "slug": "js",
      "displayName": "Javascript"
    }
  ]
}
