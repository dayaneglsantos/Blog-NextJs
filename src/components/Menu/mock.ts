import { MenuType } from '.'

export default {
  links: [
    {
      link: '/',
      text: 'Home',
      newTab: true
    },
    {
      link: '/about',
      text: 'About',
      newTab: true
    },
    {
      link: '/posts',
      text: 'Posts',
      newTab: true
    },
    {
      link: '/fallow-us',
      text: 'Fallow us',
      newTab: true
    },
    {
      link: '/contact',
      text: 'Contact',
      newTab: true
    }
  ]
} as MenuType
