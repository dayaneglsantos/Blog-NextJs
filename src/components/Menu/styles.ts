'use client'
import theme from '../../Styles/theme'
import styled from 'styled-components'

export const MenuContainer = styled.nav`
  width: 100%;
  position: sticky;
  top: 0;
  left: 0;
  padding: 14px 150px;
  background-color: ${({ theme }) => theme.colors.secondaryBackground};
  z-index: 1;

  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 5px;
    background: linear-gradient(
      to bottom,
      rgba(255, 255, 255, 0),
      rgba(0, 139, 139, 1)
    );
  }
  @media (max-width: ${theme.breakpoints.tablet}) {
    display: none;
  }
`

export const MenuLinks = styled.ul`
  display: flex;
  justify-content: center;
  gap: 48px;
  list-style: none;
`
