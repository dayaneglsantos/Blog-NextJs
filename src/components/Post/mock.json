{
  "title": "Post 2",
  "slug": "post-2",
  "alternativeText": "Foto 2",
  "image": "https://res.cloudinary.com/dbmuc0tzr/image/upload/v1703092587/img5_565f978262.jpg",
  "excerpt": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero ratione quisquam alias doloribus veniam dolore",
  "textContent": "<p>Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, commodo vitae, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. Donec non enim in turpis pulvinar facilisis. Ut felis. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus</p><ul><li>Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.</li><li>Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus.</li><li>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.</li><li>Pellentesque fermentum dolor. Aliquam quam lectus, facilisis auctor, ultrices ut, elementum vulputate, nunc.</li></ul>",
  "allowComments": true,
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "cover": "/cover.com.br/img",
  "altText": "Alt text do cover",
  "id": "1",

  "categories": [
    {
      "displayName": "React",
      "slug": "react"
    },
    {
      "displayName": "Front-end",
      "slug": "fron-end"
    }
  ],
  "tags": [
    {
      "displayName": "Tag 1",
      "slug": "tag-1"
    },
    {
      "displayName": "Tag 2",
      "slug": "tag-2"
    }
  ],
  "author": {
    "displayName": "Dayane Santos",
    "slug": "dayane-santos"
  }
}
