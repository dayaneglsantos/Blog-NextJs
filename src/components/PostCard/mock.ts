import { data } from '../../Api/data.json'

export default data.posts.data[0].attributes
