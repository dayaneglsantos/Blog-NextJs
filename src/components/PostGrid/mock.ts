import { PostGridType } from '.'

export default {
  posts: [
    {
      id: '1',
      slug: 'slug-1',
      excerpt:
        "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text.",
      title: 'Neque porro quisquam est qui dolorem',
      cover:
        'https://images.unsplash.com/photo-1682687982502-1529b3b33f85?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '2',
      slug: 'slug-2',
      excerpt:
        'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using.',
      title: 'There are many variations of passages of Lorem',
      cover:
        'https://images.unsplash.com/photo-1703877048317-c74856d65c42?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '3',
      slug: 'slug-3',
      excerpt:
        'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from de Finibus Bonorum et Malorum by Cicero are also reproduced their exact original form, accompanied by English versions from the 1914 translation by H. Rackham.',
      title: 'Sed ut perspiciatis unde omnis iste natus error',
      cover:
        'https://images.unsplash.com/photo-1703701008461-ce64767019d3?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '1',
      slug: 'slug-1',
      excerpt:
        "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text.",
      title: 'Neque porro quisquam est qui dolorem',
      cover:
        'https://images.unsplash.com/photo-1682687982502-1529b3b33f85?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '2',
      slug: 'slug-2',
      excerpt:
        'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using.',
      title: 'There are many variations of passages of Lorem',
      cover:
        'https://images.unsplash.com/photo-1703877048317-c74856d65c42?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '3',
      slug: 'slug-3',
      excerpt:
        'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from de Finibus Bonorum et Malorum by Cicero are also reproduced their exact original form, accompanied by English versions from the 1914 translation by H. Rackham.',
      title: 'Sed ut perspiciatis unde omnis iste natus error',
      cover:
        'https://images.unsplash.com/photo-1703701008461-ce64767019d3?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    }
  ],
  disabled: false,
  noMorePosts: false,
  onclick: jest.fn()
} as PostGridType
