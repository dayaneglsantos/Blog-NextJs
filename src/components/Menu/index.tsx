'use client'

import MenuMobile from '../MenuMobile'
import { MenuContainer, MenuLinks } from './styles'
import MenuLink, { MenuLinkProps } from '../MenuLink'
import Theme from '../Theme'
import { LogoProps } from '../Logo'

export type MenuType = {
  links: MenuLinkProps[]
  logo: LogoProps
}

const Menu = ({ links, logo }: MenuType) => {
  return (
    <>
      <MenuContainer data-testid="menu">
        <MenuLinks>
          {links.map((link, index) => (
            <li key={index}>
              <MenuLink
                link={link.link}
                newTab={link.newTab}
                text={link.text}
              />
            </li>
          ))}
        </MenuLinks>
        <Theme />
      </MenuContainer>

      <MenuMobile links={links} logo={logo} />
    </>
  )
}

export default Menu
