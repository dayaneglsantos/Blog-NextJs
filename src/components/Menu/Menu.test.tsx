import { fireEvent, screen } from '@testing-library/react'
import Menu from '.'
import mock from './mock'
import { renderTheme } from '../../Utils/render-theme'

describe('<Menu />', () => {
  it('should render menu correctly', () => {
    renderTheme(<Menu {...mock} />)

    const links = screen.getAllByRole('listitem')
    expect(links).toHaveLength(5)

    const btnDark = screen.getByRole('img', { name: 'Dark Theme' })
    fireEvent.click(btnDark)

    const nav = screen.getByRole('navigation')
    expect(nav).toHaveStyleRule('background-color', '#43766C')

    const btnLight = screen.getByRole('img', { name: 'Light Theme' })
    fireEvent.click(btnLight)

    expect(nav).toHaveStyleRule('background-color', '#AAD9BB')
  })
  it('should match snapshot', () => {
    const { container } = renderTheme(<Menu {...mock} />)
    expect(container).toMatchSnapshot()
  })
})
