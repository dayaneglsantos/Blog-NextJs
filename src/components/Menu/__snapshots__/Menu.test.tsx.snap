// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshot 1`] = `
.c2 {
  font-weight: bold;
  padding: 8px 12px;
  transition: all 0.3s ease-in-out;
  position: relative;
  cursor: pointer;
  filter: blur(0.5px);
  text-decoration: none;
  color: #198c8c;
}

.c2:hover {
  filter: brightness(1);
  filter: blur(0);
}

.c2::after,
.c2::before {
  content: '';
  position: absolute;
  width: 8px;
  height: 1px;
  background-color: #141010;
  transition: transform 0.3s ease-out;
}

.c2::before {
  top: 50%;
  right: 0;
  transform: scaleX(0);
  transform-origin: right;
}

.c2::after {
  bottom: 50%;
  left: 0;
  transform: scaleX(0);
  transform-origin: left;
}

.c2>span::before,
.c2>span::after {
  content: '';
  position: absolute;
  width: 1px;
  height: 8px;
  background-color: #141010;
  transition: transform 0.3s ease-out;
}

.c2>span::before {
  left: 50%;
  bottom: 0;
  transform: scaleY(0);
  transform-origin: bottom;
}

.c2>span::after {
  right: 50%;
  top: 0;
  transform: scaleY(0);
  transform-origin: top;
}

.c2:hover::after,
.c2:hover::before,
.c2:hover>span::after,
.c2:hover>span::before {
  transform: scaleX(1) scaleY(1);
}

.c3 {
  display: flex;
  gap: 8px;
  position: absolute;
  right: 8px;
  top: 25%;
  margin-right: 12px;
}

.c4 {
  border: none;
  border-radius: 50%;
  background-color: transparent;
  cursor: pointer;
}

.c5 {
  display: none;
  position: fixed;
  top: 12px;
  right: 12px;
  z-index: 2;
}

.c5 .burger {
  position: relative;
  width: 40px;
  height: 25px;
  background: transparent;
  cursor: pointer;
  display: block;
}

.c5 .burger input {
  display: none;
}

.c5 .burger span {
  display: block;
  position: absolute;
  height: 4px;
  width: 100%;
  border-radius: 9px;
  opacity: 1;
  left: 0;
  transform: rotate(0deg);
  transition: 0.25s ease-in-out;
  background-color: #198c8c;
}

.c5 .burger span:nth-of-type(1) {
  top: 0px;
  transform-origin: left center;
}

.c5 .burger span:nth-of-type(2) {
  top: 50%;
  transform: translateY(-50%);
  transform-origin: left center;
}

.c5 .burger span:nth-of-type(3) {
  top: 100%;
  transform-origin: left center;
  transform: translateY(-100%);
}

.c5 .burger input:checked~span:nth-of-type(1) {
  transform: rotate(45deg);
  top: 0px;
  left: 5px;
}

.c5 .burger input:checked~span:nth-of-type(2) {
  width: 0%;
  opacity: 0;
}

.c5 .burger input:checked~span:nth-of-type(3) {
  transform: rotate(-45deg);
  top: 28px;
  left: 5px;
}

.c0 {
  width: 100%;
  position: sticky;
  top: 0;
  left: 0;
  padding: 14px 150px;
  background-color: #AAD9BB;
  z-index: 1;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 5px;
  background: linear-gradient( to bottom,rgba(255,255,255,0),rgba(0,139,139,1) );
}

.c1 {
  display: flex;
  justify-content: center;
  gap: 48px;
  list-style: none;
}

@media (max-width:768px) {
  .c5 {
    display: flex;
  }
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

<div>
  <nav
    class="c0"
    data-testid="menu"
  >
    <ul
      class="c1"
    >
      <li>
        <a
          class="c2"
          href="/"
          target="_blank"
        >
          Home
          <span />
        </a>
      </li>
      <li>
        <a
          class="c2"
          href="/about"
          target="_blank"
        >
          About
          <span />
        </a>
      </li>
      <li>
        <a
          class="c2"
          href="/posts"
          target="_blank"
        >
          Posts
          <span />
        </a>
      </li>
      <li>
        <a
          class="c2"
          href="/fallow-us"
          target="_blank"
        >
          Fallow us
          <span />
        </a>
      </li>
      <li>
        <a
          class="c2"
          href="/contact"
          target="_blank"
        >
          Contact
          <span />
        </a>
      </li>
    </ul>
    <div
      class="c3"
    >
      <button
        class="c4"
      >
        <img
          alt="Dark Theme"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
      </button>
      <button
        class="c4"
      >
        <img
          alt="Light Theme"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
      </button>
    </div>
  </nav>
  <div>
    <div
      class="c5"
    >
      <label
        class="burger"
        for="burger"
      >
        <input
          aria-label="Menu burger"
          id="burger"
          type="checkbox"
        />
        <span />
        <span />
        <span />
      </label>
    </div>
  </div>
</div>
`;
